include/swift/AST/Pattern.h

```diff
@@ -722,17 +723,16 @@ class VarPattern : public Pattern {
   void setSubPattern(Pattern *p) { SubPattern = p; }
 
   static bool classof(const Pattern *P) {
-    return P->getKind() == PatternKind::Var;
+    return P->getKind() == PatternKind::Binding;
   }
 };
 
-
 inline Pattern *Pattern::getSemanticsProvidingPattern() {
   if (auto *pp = dyn_cast<ParenPattern>(this))
     return pp->getSubPattern()->getSemanticsProvidingPattern();
   if (auto *tp = dyn_cast<TypedPattern>(this))
     return tp->getSubPattern()->getSemanticsProvidingPattern();
-  if (auto *vp = dyn_cast<VarPattern>(this))
+  if (auto *vp = dyn_cast<BindingPattern>(this))
```

Comment:
Nit: Should these variable names be updated as well (in this case, from `vp` to `bp` for example)?

Reply:
I want to make this a pure refactor change. I am fine going back through later and fixing.